src/commons/services/httpInterceptor.js

```diff
@@ -0,0 +1,48 @@
+import {Store} from "aurelia-store";
+import {inject} from "aurelia-dependency-injection";
+import {setError} from "../../store/store";
+import { Router } from "aurelia-router";
+
+@inject(Store, Router)
+export class HttpInterceptor {
+
+  constructor(store, router) {
+    this.router = router;
+    this.store = store;
+    this.store.registerAction('setError', setError);
+  }
+
+  async response(response) {
+    if (!response.ok) {
+      if (response.status === 502) {
```

Comment:
maybe additional logic could be added to handle 401 and 403 responses. for example redirect to login page?
If I understand correctly, then we get 403 in case of token expiration and unknown user/insufficient privileges.